doc/mgmt/SONiC_DNS_Support.md

```diff
@@ -0,0 +1,433 @@
+
+### 3.2.1 CONFIG DB
+```
+; DNS configuration attributes global to the system. Only one instance of the table exists in the system.
+; Key
+global_key           = “global”  ;  DNS global configuration
+; Attributes
+src_intf               = ifname  ;  source interface address for the outgoing DNS packets
+
+```
+
+```
+; DNS name server configuration in the system.
+; Key
+server_key           = IPAddress  ;  DNS server’s address
+; Attributes
+No attributes are introduced as part of this design.
+```
+
+### 3.2.2 APP DB
+
+### 3.2.3 STATE DB
+
+### 3.2.4 ASIC DB
+
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+
+YANG models needed for DNS handling in the management framework:
+1. **openconfig-system.yang**  
+
+2. **sonic-system-dns.yang**
+
+Supported yang objects and attributes:
+```diff
+
+module: openconfig-system
+      +--rw system
+
+      +--rw dns
+           |  +--rw config
+           |  |  +--rw search*                      oc-inet:domain-name
+           |  |  +--rw oc-sys-ext:source-intf?   -> /oc-if:interfaces/interface/name
+           |  +--ro state
+           |  |  +--ro search*                      oc-inet:domain-name
+           |  |  +--ro oc-sys-ext:source-intf?   -> /oc-if:interfaces/interface/name
+           |  +--rw servers
+           |  |  +--rw server* [address]
+           |  |     +--rw address    -> ../config/address
+           |  |     +--rw config
+           |  |     |  +--rw address?   oc-inet:ip-address
+           |  |     |  +--rw port?      oc-inet:port-number
+           |  |     +--ro state
+           |  |        +--ro address?   oc-inet:ip-address
+           |  |        +--ro port?      oc-inet:port-number
+
+The above "source-intf" & "address" fields are supported in the scope of this work.
+
+module: sonic-system-dns
+  +--rw sonic-system-dns
+     +--rw DNS_SERVER
+     |  +--rw DNS_SERVER_LIST* [ipaddress]
+     |     +--rw ipaddress    inet:ip-address
+     +--rw DNS
+        +--rw DNS_LIST* [type]
+           +--rw type      enumeration ('global')
+           +--rw src_intf?   union
+
+```
+
+### 3.6.2 CLI
+
+
+#### 3.6.2.1 Configuration Commands
+All commands are executed in `configuration-view`:
+```
+sonic# configure terminal
+sonic(config)#
+```
+
+##### 3.6.2.1.1 Configure DNS server & source IP
+```
+sonic(config)# ip name-server
+  source-intf   Configure source interface to pick the source IP, used for the DNS query
+  A.B.C.D/A::B  Domain name server
+
+sonic(config)# ip name-server source-intf
+    Ethernet     Ethernet interface
+    Loopback     Loopback interface
+    Management   Management interface
+    PortChannel  PortChannel interface
+    Vlan         Vlan interface
+
+sonic(config)# ip name-server 10.11.0.1
+
+sonic(config)# ip name-server 2001:aa:aa::a
+
+sonic(config)# ip name-server source-intf Loopback 0

```

Comment:
This mean, all name-server will use the same source-interface. Should take this parameter per server i.e. ip name-server ip-address source-intf Loopback0